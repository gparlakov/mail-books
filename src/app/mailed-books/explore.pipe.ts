import { Pipe } from "@angular/core";

@Pipe({
    name:'explore',
    pure: false
})
export class ExplorePipe {
    transform(value: any) {
        console.log(value)
    }
}
