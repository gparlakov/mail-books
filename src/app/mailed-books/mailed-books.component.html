<mat-selection-list #books >
    <mat-list-item>
        <button mat-button (click)="books.selectAll()">select all</button>
        <button mat-button (click)="books.deselectAll()">deselect all</button>
        <button mat-flat-button color="accent">download</button>
    </mat-list-item>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name - Author  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name 1 - Author 1  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name 2 - Author 2  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name - Author  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name 1 - Author 1  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name 2 - Author 2  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name - Author  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name 1 - Author 1  </span>
    </mat-list-option>
    <mat-list-option [checkboxPosition]="before">
        <img src="/favicon.ico" width="25" height="25"/>
        <span>Book name 2 - Author 2  </span>
    </mat-list-option>
</mat-selection-list>

