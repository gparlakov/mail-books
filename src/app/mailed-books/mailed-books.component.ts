import { Component, OnInit, ViewChild } from '@angular/core';
import { MatSelectionList } from '@angular/material/list';
import { SharedBooksService } from './shared-books.service';

@Component({
    selector: 'mbs-mailed-books',
    templateUrl: './mailed-books.component.html',
    styleUrls: ['./mailed-books.component.scss']
})
export class MailedBooksComponent implements OnInit {

    @ViewChild('books')
    private books: MatSelectionList;

    bookItems: SharedBook[];

    constructor(private sharedBooks: SharedBooksService) { }

    ngOnInit() {
        this.bookItems = this.sharedBooks.getBy();
    }

    selectAll() {
        this.books.selectAll();
    }
}

export class SharedBook {
    constructor(
        public readonly title: string,
        public readonly author: string,
        public readonly sharedBy: string,
    ){

    }
}


