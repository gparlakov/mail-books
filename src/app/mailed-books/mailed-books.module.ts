import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MailedBooksComponent } from './mailed-books.component';
import { MatListModule } from '@angular/material/list';
@NgModule({
    imports: [
        CommonModule,
        MatListModule
    ],
    declarations: [MailedBooksComponent],
    exports: [MailedBooksComponent]
})
export class MailedBooksModule { }
