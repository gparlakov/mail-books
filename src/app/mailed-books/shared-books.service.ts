import { Injectable } from '@angular/core';
import { SharedBook } from './mailed-books.component';

@Injectable()
export class SharedBooksService {

    getBy(): SharedBook[] {
        // throw new Error("Method not implemented.");
        return [];
    }
}
