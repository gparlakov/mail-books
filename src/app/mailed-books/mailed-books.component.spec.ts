import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { MailedBooksComponent } from './mailed-books.component';
import { MatListModule } from '@angular/material/list';
import { MatButtonModule } from '@angular/material/button';
import { NO_ERRORS_SCHEMA, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';

describe('MailedBooksComponent', () => {
  let component: MailedBooksComponent;
  let fixture: ComponentFixture<MailedBooksComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ MailedBooksComponent ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    //   imports: [MatButtonModule, MatListModule]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MailedBooksComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.elementRef.nativeElement).toMatchSnapshot();

  });
});
