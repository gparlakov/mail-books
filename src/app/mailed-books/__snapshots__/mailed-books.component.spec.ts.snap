// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailedBooksComponent should create 1`] = `
<div
  id="root0"
  ng-version="6.0.7"
>
  <mat-selection-list
    aria-disabled="false"
    class="mat-selection-list"
    role="listbox"
    tabindex="0"
  >
    <mat-list-item
      class="mat-list-item"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="true"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <div
          class="mat-list-text"
        />
        <button
          class="mat-button"
          mat-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            select all
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <button
          class="mat-button"
          mat-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            deselect all
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <button
          class="mat-flat-button mat-accent"
          color="accent"
          mat-flat-button=""
          ng-reflect-color="accent"
        >
          <span
            class="mat-button-wrapper"
          >
            download
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
      </div>
    </mat-list-item>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name - Author 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name 1 - Author 1 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name 2 - Author 2 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name - Author 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name 1 - Author 1 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name 2 - Author 2 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name - Author 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name 1 - Author 1 
          </span>
        </div>
      </div>
    </mat-list-option>
    <mat-list-option
      aria-disabled="false"
      aria-selected="false"
      class="mat-list-item mat-list-option"
      role="option"
      tabindex="-1"
    >
      <div
        class="mat-list-item-content"
      >
        <div
          class="mat-list-item-ripple mat-ripple"
          mat-ripple=""
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLUnknownElement]"
        />
        <mat-pseudo-checkbox
          class="mat-pseudo-checkbox"
          ng-reflect-disabled="false"
          ng-reflect-state="unchecked"
        />
        <div
          class="mat-list-text"
        >
          <img
            height="25"
            src="/favicon.ico"
            width="25"
          />
          <span>
            Book name 2 - Author 2 
          </span>
        </div>
      </div>
    </mat-list-option>
  </mat-selection-list>
</div>
`;
